solver/src/solver/paraswap_solver.rs

```diff
@@ -162,6 +152,16 @@ impl ParaswapSolver {
         Ok(Some(settlement))
     }
 
+    fn account(&self) -> &Account {
+        &self.account
+    }
+
+    fn name(&self) -> &'static str {
+        "ParaSwap"
+    }
+}
+
+impl ParaswapSolver {
```

Comment:
I think the ParaswapSolver should also have it's settle_order function removed because it duplicates the same function from the trait again.
Furthermore I feel we should remove the whole `fn settle_order(&self, order: LimitOrder) -> Result<Option<Settlement>>` from the trait and instead move it to `struct SingleOrderSolver`. This way it is not possible to overwrite the function in trait implementors which would have caught the ParaswapSolver part I'm commenting on.

Reply:
This is a great suggestion, although I am running into the issue that inside the `solve` method defined in 

```
impl<I: SingleOrderSolving + Send + Sync> Solver for SingleOrderSolver<I>
```

there is "no method named `settle_order` found for reference `&SingleOrderSolver<I>`"

```
tokio::time::timeout_at(deadline.into(), self.settle_order(order))
   |                                          ^^^^^^^^^^^^ method not found in `&SingleOrderSolver<I>`
```